ratpack-core/src/main/java/ratpack/http/client/internal/HttpChannelKey.java

```diff
@@ -17,25 +17,33 @@
 package ratpack.http.client.internal;
 
 import io.netty.channel.EventLoop;
+import ratpack.api.Nullable;
 import ratpack.exec.ExecController;
 import ratpack.exec.Execution;
 
 import java.net.URI;
 import java.time.Duration;
+import java.util.Objects;
 
 final class HttpChannelKey {
 
   final boolean ssl;
   final int port;
   final String host;
 
+  final ProxyInternal proxy;
```

Comment:
The ProxyInternal has an SslContext property that is a copy of the SslContext set on a RequestSpec.  This SslContext property is not 

However, the SslContext is _not_ used in equality checks. SslContext does not implement equals and hashcode, and writing a Comparator to do so or similar would be complex. Instead, Proxy.protocol is the only new property that influences equality.

This means that some request-level changes to the SslContext will _not_ result in a new connection being established with the proxy. There are some rare use cases that this might create problems for (such as if SSL client authentication is in use, and different client certificates result in different proxy behavior), but these are consistent with Ratpack's HttpCient SSL support in general (direct connections also have this same problem).

The SslContext is only saved on a ProxyInternal (and therefore HttpChannelKey) if the user
has configured an HTTP Proxy and has set the Proxy protocol to HTTPS. This eliminates the memory increase for the common case where proxies are not used, or when SSL is not used to communicate to the Proxy.